packages/tools/kadena-cli/src/account/tests/accountDetails.test.ts

```diff
@@ -0,0 +1,182 @@
+        () => {
+          return HttpResponse.json(
+            { error: 'with-read: row not found: qwerty' },
+            { status: 404 },
+          );
+        },
+      ),
+    );
+    const res = await runCommand(
+      'account details --account=account1 --fungible=coin --network=testnet --chain-ids=2',
+    );
+
+    expect(res.stderr).toContain(
+      '\nAccount "account1" is not available on\nfollowing chain(s): 2 on network "testnet04"',
+    );
+  });
+
+  it('should return account details for multiple chain ids', async () => {
+    const chainIdOneData = {
+      ...accountDetailsSuccessData,
+      result: {
+        ...accountDetailsSuccessData.result,
+        data: {
+          ...accountDetailsSuccessData.result.data,
+          balance: 100,
+        },
+      },
+    };
+    server.use(
+      http.post(
+        'https://api.testnet.chainweb.com/chainweb/0.0/testnet04/chain/1/pact/api/v1/local',
+        async (): Promise<HttpResponse> => {
+          return HttpResponse.json(chainIdOneData, { status: 200 });
+        },
+      ),
+    );
+    mockPrompts({
+      select: {
+        'Select an account (alias - account name):': 'accountName',
+        'Select a network:': 'testnet',
+      },
+      input: {
+        'Enter the name of a fungible:': 'coin',
+        'Enter a ChainId (0-19) (comma or hyphen separated e.g 0,1,2 or 1-5 or all):':
+          '0,1',
+      },
+    });
+
+    const res = await runCommandJson('account details');
+    expect(res).toEqual({
+      ['0']: accountDetailsSuccessData.result.data,
+      ['1']: chainIdOneData.result.data,
+    });
+  });
+
+  it('should return account details for one chain id and for two chain ids account not found', async () => {
+    server.use(
+      http.post(
+        'https://api.testnet.chainweb.com/chainweb/0.0/testnet04/chain/1/pact/api/v1/local',
+        async (): Promise<HttpResponse> => {
+          return HttpResponse.json(
+            { error: 'with-read: row not found: qwerty' },
+            { status: 404 },
+          );
+        },
+      ),
+    );
+    server.use(
+      http.post(
+        'https://api.testnet.chainweb.com/chainweb/0.0/testnet04/chain/2/pact/api/v1/local',
+        async (): Promise<HttpResponse> => {
+          return HttpResponse.json(
+            { error: 'with-read: row not found: qwerty' },
+            { status: 404 },
+          );
+        },
+      ),
+    );
+    const res = await runCommandJson(
+      'account details --account=account1 --fungible=coin --network=testnet --chain-ids=0-2',
+    );
+    expect(res).toEqual({
+      ['0']: accountDetailsSuccessData.result.data,
+    });
+  });
+
+  it('should return error message when api fails', async () => {
+    server.use(
+      http.post(
+        'https://api.testnet.chainweb.com/chainweb/0.0/testnet04/chain/2/pact/api/v1/local',
+        () => {
+          return HttpResponse.json('shit hit the fan', { status: 500 });
```

Comment:
is this a response we actually get back from chainweb and u mocked :S lol

Reply:
lol, no :D it could be any error message, so added this